src/datatype/casts.rs

```diff
@@ -83,3 +84,35 @@ fn _vectors_cast_svecf32_to_vecf32(
 
     Vecf32Output::new(Vecf32Borrowed::new(&slice))
 }
+
+#[pgrx::pg_extern(immutable, parallel_safe, strict)]
+fn _vectors_cast_vecf32_to_bvecf32(
+    vector: Vecf32Input<'_>,
+    _typmod: i32,
+    _explicit: bool,
+) -> BVecf32Output {
+    let mut values = BVecf32Owned::new_zeroed(vector.len() as u16);
+    for (i, &x) in vector.slice().iter().enumerate() {
+        match x.to_f32() {
+            x if x == 0. => {}
+            x if x == 1. => values.set(i, true),
```

Comment:
Shall we change it to `1 if  x>0`? @usamoi

Reply:
If we want to accept errors, we could use `EPS`. But I think it's not necessary here.

Reply:
Currently only [0,1,1] vector works. But we need the casting to do the binary quantization. I personally think it's fine to let element bigger than zero to be 1, and other to be 0. What do you think?

Reply:
We can provide a function to do it. It's more explicit.

Reply:
And should not be == 0. There are some values below 0.

e.g. [ -0.022388368844985962, 0.028537113219499588, -0.017491688951849937, -0.01533466950058937, -0.03451230376958847, 0.0162644200026989, -0.020231351256370544, -0.012743767350912094, -0.032082557678222656, 0.0037468906957656145, -0.003864658996462822, 0.004292343743145466, 0.007035105023533106, -0.03314867243170738, ...]

Reply:
completed in new commit